mantidproject/mantid
# Basic math operations for Histogram

This PR adds a few basic math operations for `Histogram` and some of the X, Y, and E types.
- `-` and `/` with `double` argument for Y and E types (previously only `+` and `*` were implemented)
- `=` with `double` argument for Y and E types.
- `*` and `/` with `double` argument for `Histogram`
- `+` and `-` for two `Histograms`.
- Where applicable, also the corresponding `+=` operators (etc.) have been added.

**To test:**
- Thorough code review.
- Check that the tests leave no holes.
- Please comment whether you think the amount of testing of input values is reasonable and correct.

This partially resolves #17137.

No release notes (we still have no Python interface for `HistogramData`).

Once code review confirmed that these changes are as we want them I will update the developer documentation with a list of newly available operations.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
